### PR TITLE
layout: Avoid double negation in `CellLayout::is_empty_for_empty_cells()`

### LINKED ISSUE #33678: Avoid double negation in `is_empty_for_empty_cells()`
https://github.com/servo/servo/blob/f8933a57353aeca14a6cbc60b3cb0cf98cab6c5d/components/layout_2020/table/layout.rs#L76-L82

Could avoid the double negation by using `all()` instead of `any()`: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.all

### PR DESCRIPTION
**Pull request by Taym95**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #33678 

- [x] These changes do not require tests because there is no behavior change.

### CODE REVIEW

**Review by mrobinson — APPROVED** [review]

File: components/layout_2020/table/layout.rs [hunk 1]
```diff
@@ -78,7 +78,7 @@ impl CellLayout {
             .layout
             .fragments
             .iter()
-            .any(|fragment| !matches!(fragment, Fragment::AbsoluteOrFixedPositioned(_)))
+            .all(|fragment| matches!(fragment, Fragment::AbsoluteOrFixedPositioned(_)))
```
Comment (Loirooriol) [inline components/layout_2020/table/layout.rs:81]: This is only removing one negation, you need to remove the one in `!self` too.
That is, checking that there isn't any of the fragments which is different than AbsoluteOrFixedPositioned is equivalent to checking that all of the fragments are AbsoluteOrFixedPositioned.

**Review by Loirooriol — CHANGES_REQUESTED** [review]

**Comment by Taym95** [comment]
> Bots caught the problem indeed: https://github.com/servo/servo/actions/runs/11216734706#summary-31178398934

Sorry I missed that, empty-cells-applies-to-014.xht is passing now

**Review by Loirooriol — APPROVED** [review]

**Comment by mrobinson** [comment]
> Bots caught the problem indeed: https://github.com/servo/servo/actions/runs/11216734706#summary-31178398934

Whoops. Sorry. I should have caught this in my review.

**Comment by Loirooriol** [comment]
Lint is failing: https://github.com/servo/servo/actions/runs/11217687511/job/31179784487?pr=33688

**Pull request merged**